pkgs/tools/misc/memtest86+/default.nix

```diff
@@ -1,31 +1,28 @@
-{ stdenv, fetchgit }:
+{ stdenv, fetchurl }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "memtest86+";
-  version = "5.01-coreboot-002";
+  version = "5.31b";
 
-  src = fetchgit {
-    url = "https://review.coreboot.org/memtest86plus.git";
-    rev = "v002";
-    sha256 = "0cwx20yja24bfknqh1rjb5rl2c0kwnppzsisg1dibbak0l8mxchk";
+  src = fetchurl {
+    url = "https://www.memtest.org/download/${version}/memtest86+-${version}.tar.gz";
+    sha256 = "028zrch87ggajlb5xx1c2ab85ggl9qldpibf45735sy0haqzyiki";
   };
 
-  NIX_CFLAGS_COMPILE = "-I. -std=gnu90";
-
   hardeningDisable = [ "all" ];
 
-  buildFlags = [ "memtest.bin" ];
-
-  doCheck = false; # fails
+  doCheck = stdenv.isi686;
+  checkTarget = "run_self_test";
 
   installPhase = ''
     install -Dm0444 -t $out/ memtest.bin
   '';
 
-  meta = {
-    homepage = "http://www.memtest.org/";
-    description = "A tool to detect memory errors";
-    license = stdenv.lib.licenses.gpl2;
+  meta = with stdenv.lib; {
+    homepage = "https://www.memtest.org/";
+    description = "An advanced memory diagnostic tool";
+    license = licenses.gpl2;
```

Comment:
plus or only?

Reply:
neither?
here's the header from `main.c`
```
/*
 * MemTest86+ V5 Specific code (GPL V2.0)
 * By Samuel DEMEULEMEESTER, sdemeule@memtest.org
 * http://www.canardpc.com - http://www.memtest.org
 * ------------------------------------------------
 * main.c - MemTest-86  Version 3.5
 *
 * Released under version 2 of the Gnu Public License.
 * By Chris Brady
 */
```
maybe that's implicitly `gpl2Only`, but since `licenses.gpl2` exists, that's what i'd go with

note: the license text is not included in the provided source archive

Reply:
> since licenses.gpl2 exists, that's what i'd go with

Since you created that PR this license got deprecated and should no be used anymore.

Reply:
i saw that, i assumed that meant one shouldn't use an ambiguous license header
but i have no authority to change their license header...

i'll proceed on the assumption that i'm right that this is implicitly `gpl2Only` and change it to that